### DISCUSSION TITLE
Having difficulty with decorators - trying to get Google Cloud Functions working

### DISCUSSION THREAD

**Discussion by nbrandaleone** [hylang/hy#2455] [Q&A] [answered]
Hello Hy community!

I am trying to get a Hy script to be properly translated into Python, so that Google Functions Framework will accept the code. Ultimately, I want to upload it to Google Cloud Functions.

The Python code *should* look like the following:
```
import functions_framework

@functions_framework.http
def hello(request):
    return 'Hello world!'
```
Unfortunately, I can't get the decorator to create the signature properly.  It ALWAYS puts a set of function parens on the end.
The best I can get is below - which does not work with the Functions-Framework Google python library:

```
import functions_framework

@functions_framework.http()
def hello(request):
    return 'Hello world!'
```

My source Hy code is below.  I am using `hy2py` for the conversion.

```
(import functions_framework)

;; Function is passed a single parameter, `request`,
;; which is a Flask REQUEST object
(defn [(.http functions-framework)] hello [request]
  "Hello world!")
```

Any help would be appreciated.  Although a long-time lisp user, I am new to Hy.

Thanks,
Nick

**Comment by Kodiologist** [comment, marked as answer]
The implicit call works the same as in Hy as in Python. You need to write `(defn [functions-framework.http] …)` or `(defn [(. functions-framework http)] …)` in Hy to get `@functions_framework.http` in Python.

Don't forget that decorators are little more than syntactic sugar for higher-order functions anyway. You can always say e.g. `(setv hello (functions-framework.http (fn [request] …)))`.

**Reply by nbrandaleone** [reply]
Thanks!  I will give it a try.

Nick

**Reply by goyalyashpal** [reply]
> I will give it a try.

did it worK?

**Reply by nbrandaleone** [reply]
It did work - thank you again!
I wrote it up in a blog article.  See: https://medium.com/google-cloud/three-lisps-on-gcp-cloud-functions-43927b98dcbd

Nick
